### ISSUE TITLE
Incompatible with react-scripts@5.0.0, quick start guide instructions do not work

### ISSUE THREAD

**Issue by monbrey** [instructure/instructure-ui#904]
### Background Information

**Package Version(s)**:
@instructure/ui v8.16.0

### Describe the Bug
Projects using this library cannot build using [Quick Start guide instructions](https://instructure.design/#usage).
This is due to an incompatibility with react-scripts@5.0.0, which is the current default shipped in create-react-app.

### Steps To Reproduce
1. Start a new project using create-react-app as per [Quick Start guide](https://instructure.design/#usage)
2. Install the @instructure/ui package
`npm i @instructure/ui`
3. Modify the default index.js file to import and use @instructure/ui, e.g:
```diff
import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import App from './App';
import reportWebVitals from './reportWebVitals';
+ import { canvas, InstUISettingsProvider } from '@instructure/ui';

ReactDOM.render(
	<React.StrictMode>
+ 		<InstUISettingsProvider theme={canvas}>
			<App />
+ 		</InstUISettingsProvider>
	</React.StrictMode>,
	document.getElementById('root')
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
```
4. Run `npm start`

### Expected Behavior
I would expect the project to build, but it won't

### Screenshots

The project will not build, due to a large number of errors.
![image](https://user-images.githubusercontent.com/5294381/152444809-eb5db00b-8ad2-4742-8131-8ee75a04b670.png)
![image](https://user-images.githubusercontent.com/5294381/152444747-ef116630-7223-466c-850d-1641bab81212.png)

### Additional Information

**Current Workaround(s):**
Downgrade to `react-scripts@4.0.3`
`npm i react-scripts@4.0.3`

**Products Affected:**
Appears to be any usage of the whole package.

**Are you willing to submit a PR to fix?**
- [X] Yes, I'm willing to submit a PR

I'd be willing to submit a PR to update the https://github.com/instructure/instructure-ui/blob/master/docs/getting-started/usage.md page with a warning or similar to notify users of this, however I don't know if there is a way to "fix" this package to work with react-scripts@5.0.0, or if react-scripts itself needs to be fixed.

**Requested Priority:**
Medium - workaround is available but users should be informed.

**Comment by matyasf** [comment]
hi
Thanks for reporting this issue! Looks like a bug to me, InstUI should work with `react-scripts` v5. We'll look into it and try to fix it

**Comment by matyasf** [comment]
We have released InstUI 8.17.0, which should get InstUI working with React-scripts 5. Actually the issue was with Webpack 5 which is used by the latest react-scripts.

https://github.com/instructure/instructure-ui/releases/tag/v8.17.0
